OptionSelector/CategoryTableViewController.m

```diff
@@ -0,0 +1,118 @@
+//
+//  CategoryTableViewController.m
+//  OptionSelector
+//
+//  Created by Henna on 8/10/15.
+//  Copyright (c) 2015 Mike Kavouras. All rights reserved.
+//
+
+#import "CategoryTableViewController.h"
+#import "CQCategory.h"
+#import "OptionsTableViewController.h"
+#import "CategoryModel.h"
+
+
+@interface CategoryTableViewController ()
+@property (nonatomic) CategoryModel * data;
+
+@end
+
+@implementation CategoryTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.title = @"Henna's Categories";
+    self.data = [CategoryModel sharedInstance];
+    [self.data initializeModel];
+    
+}
+
+- (void)viewDidAppear:(BOOL)animated {
+[self.tableView reloadData];
+}
+
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return [self.data.allCategories count];
+}
+
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"CategoryCellIdentifier"forIndexPath:indexPath];
+    
+    CQCategory * category = [self.data.allCategories objectAtIndex:indexPath.row];
+    
+    NSString * title = category.name;
+    NSString * selected = category.selection;
+    cell.textLabel.text = title;
+
+    cell.detailTextLabel.text = selected;
+    
+    return cell;
+}
+
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    NSIndexPath * indexPath = [self.tableView indexPathForSelectedRow];
+    CQCategory * selected = [self.data.allCategories objectAtIndex:indexPath.row];
+    OptionsTableViewController * destination = segue.destinationViewController;
+    destination.category = selected;
+}
+
+
+/*
```

Comment:
Style: You should remove these comments to make your code more compact and readable.